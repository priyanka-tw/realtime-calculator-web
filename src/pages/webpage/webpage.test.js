import React from 'react';
import {render} from '@testing-library/react';
import 'jest-styled-components';
import Context from "../../services/global-context-provider/context";
import Webpage from "./webpage";

describe('Webpage', () => {

    it('should render login page if the user is not logged in', () => {
        const {container} = render(<Context.Provider value={{username: '', setSocket: jest.fn()}}>
            <Webpage/> </Context.Provider>)

        expect(container).toMatchSnapshot();
    });

    it('should render home page if the user is logged in', () => {
        const {container} = render(<Context.Provider value={{username: 'test-username'}}>
            <Webpage/> </Context.Provider>)

        expect(container).toMatchSnapshot();
    });
});
