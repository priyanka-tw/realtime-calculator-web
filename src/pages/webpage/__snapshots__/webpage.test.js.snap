// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpage should render home page if the user is logged in 1`] = `
<div>
  <p>
    Home Page 
  </p>
   
</div>
`;

exports[`Webpage should render login page if the user is not logged in 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="c0"
  />
   
</div>
`;
